app/models/middleware_server.rb

```diff
@@ -33,6 +37,18 @@ def evaluate_alert(alert_id, event)
   end
 
   def in_domain?
-    !middleware_server_group.nil?
+    middleware_server_group
+  end
+
+  def hawkular?
+    product == 'Hawkular'
+  end
+
+  def immutable?
+    hawkular? || properties['Immutable'] == 'true'
```

Comment:
This should really be implemented by the subclass instead of checking `hawkular?` here no?  Seeing `if hawkular?` on a base model feels like it is breaking the pluggability (/cc @durandom )

Why not have `immutable?` return `false` on the base model and have `ManageIQ::Providers::Hawkular::MiddlewareManager::MiddlewareServer#immutable?` return `properties['Immutable'] == 'true'`